### ISSUE TITLE
Feature for annotation inference

### ISSUE THREAD

**Issue by Ao-senXiong** [eisopux/checker-framework-vscode#151]
Recently, I have used Intellij's nullness analysis system and it seems they can infer the annotations automatically.

It would be good if we can have the inference system integrated with our extension.

**Comment by wmdietl** [comment]
Yes, that would be great! I think this is probably better as separate `checker-framework-inference-languageserver` and `checker-framework-inference-vscode` projects.
